service/controller/resource/nodepool/template/parameters.go

```diff
@@ -0,0 +1,209 @@
+package template
+
+import (
+	"strconv"
+
+	azureresource "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+	"github.com/giantswarm/microerror"
+	"sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+)
+
+type Parameters struct {
+	AzureOperatorVersion string
+	ClusterID            string
+	DataDisks            []v1alpha3.DataDisk
+	NodepoolName         string
+	OSImage              OSImage
+	Scaling              Scaling
+	SSHPublicKey         string
+	SubnetName           string
+	VMCustomData         string
+	VMSize               string
+	VnetName             string
+	Zones                []string
+}
+
+type Scaling struct {
+	MinReplicas     int32
+	MaxReplicas     int32
+	CurrentReplicas int32
+}
+
+type OSImage struct {
+	Publisher string
+	Offer     string
+	SKU       string
+	Version   string
+}
+
+func (p Parameters) ToDeployParams() map[string]interface{} {
+	allParams := map[string]interface{}{}
+	allParams["azureOperatorVersion"] = struct {
+		Value interface{}
+	}{
+		Value: p.AzureOperatorVersion,
+	}
```

Comment:
```suggestion
func param(v interface{}) interface{} {
	return struct{
		Value interface{}
	}{
		Value: v,
	}
}

func (p Parameters) ToDeployParams() map[string]interface{} {
	allParams := map[string]interface{}{}
	allParams["azureOperatorVersion"] = param(p.AzureOperatorVersion)
```

Would it make any sense to do this sort of indirection to slightly improve readability / maintainability of parameters?

Reply:
Yeah, I think it's a good idea.